src/hardware/serialport/serialport.cpp

```diff
@@ -1308,6 +1308,8 @@ class SERIALPORTS:public Module_base {
 	}
 
 	~SERIALPORTS () {
+		MODEM_ClearPhonebook();
```

Comment:
Complete nitpick: pedantically speaking, this line should be placed after the `for` loop (so we keep the reverse order compared to initialization).

However, in this case it doesn't matter, `serialports` does not depend on data cleaned in `MODEM_ClearPhonebook()` at all, and likely never will.

Reply:
And I suppose if @NicknineTheEagle moves from using pointers to objects like you've nicely suggested @dreamer, we can let the phonebook container clean itself up.